configuration/decode_clusters_whitebox_test.go

```diff
@@ -0,0 +1,112 @@
+package configuration
+
+import (
+	"testing"
+
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/fabric8-services/fabric8-cluster/cluster/repository"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/fabric8-services/fabric8-common/resource"
+)
+
+func TestDecodeClusters(t *testing.T) {
+
+	resource.Require(t, resource.UnitTest)
+
+	t.Run("ok", func(t *testing.T) {
+		// given
+		data := []map[string]interface{}{
+			{
+				"name":                            "cluster1",
+				"api-url":                         "http://cluster1",
+				"console-url":                     "http://console.cluster1",
+				"logging-url":                     "http://logging.cluster1",
+				"metrics-url":                     "http://metrics.cluster1",
+				"app-dns":                         "appdns1",
+				"auth-client-id":                  "authclientID1",
+				"auth-client-secret":              "authclientsecret1",
+				"auth-client-default-scope":       "defaultscope1",
+				"capacity-exhausted":              true,
+				"service-account-username":        "udername1",
+				"service-account-token":           "token1",
+				"service-account-token-encrypted": true,
+				"token-provider-id":               "provider1",
+				"type":                            "OSD",
+			},
+			{
+				"name":                      "cluster2",
+				"api-url":                   "http://cluster2",
+				"app-dns":                   "appdns2",
+				"auth-client-id":            "authclientID2",
+				"auth-client-secret":        "authclientsecret2",
+				"auth-client-default-scope": "defaultscope2",
+				"service-account-username":  "udername2",
+				"service-account-token":     "token2",
+				"token-provider-id":         "provider2",
+			},
+		}
+
+		// when
+		result, err := decodeClusters(data)
+		// then
+		require.NoError(t, err)
+		require.Len(t, result, 2)
+		// a cluster with all optional fields explicitely set
```

Comment:
typo `explicitly`

Reply:
one day, I'll stop making this typo...

Reply:
no worries, it happens

Reply:
yeah, I guess it's because in French we write `explicitement`, so I tend to start with `explicite` in English as well... But I can fix it :D

Reply:
fixed in 4320994